components/server/src/websocket/websocket-connection-manager.ts

```diff
@@ -435,10 +434,11 @@ class GitpodJsonRpcProxyFactory<T extends object> extends JsonRpcProxyFactory<T>
                         message: e.message,
                     },
                 );
+                throw new ResponseError(e.code, e.message);
```

Comment:
I assume this is the cause for https://linear.app/gitpod/issue/WEB-613.
We're missing the optional `data` attribute from `ApplicationError` on client side because it's not propagated here. 

Testing a fix here: https://github.com/gitpod-io/gitpod/pull/18304/commits/3dc98a959866dc84471783008f7629f21c620228